packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts

```diff
@@ -409,7 +509,28 @@ export class FileSystem extends FileSystemBase {
       throughputMode: props.throughputMode,
       provisionedThroughputInMibps: props.provisionedThroughputPerSecond?.toMebibytes(),
       backupPolicy: props.enableAutomaticBackups ? { status: 'ENABLED' } : undefined,
-      fileSystemPolicy: Lazy.any({ produce: () => this._fileSystemPolicy }),
+      fileSystemPolicy: Lazy.any({
+        produce: () => {
+          const denyAnonymousAccessFlag = FeatureFlags.of(this).isEnabled(cxapi.EFS_DENY_ANONYMOUS_ACCESS) ?? false;
+          const denyAnonymousAccessByDefault = denyAnonymousAccessFlag || this._grantedClient;
```

Comment:
help me understand why `this._grantedClient` is included here?

Reply:
Originally, my design was to reject anonymous clients only when using `grantXXX` (e.g. `grantRead`). Currently, an implementation that rejects anonymous clients by default based on feature flags has been added. Therefore, it is also possible to remove `_grantedClient` and switch to feature flags only.
However, even if a user who has upgraded the CDK from an older version starts using `grantXXX`, anonymous clients will not be rejected unless the feature flag is enabled, so I implemented an implementation that determines the permission of an anonymous client with both the feature flag and `_grantedClient` according to the original purpose. Nonetheless, I'm going to try not to use `_grantedClient` if it's unnecessary because it affects maintainability. What do you think?